src/sideBar.ts

```diff
@@ -0,0 +1,767 @@
+import * as vscode from 'vscode';
+import * as nls from 'vscode-nls';
+import CMakeProject from './cmakeProject';
+import * as preset from './preset';
+import { runCommand } from './util';
+
+nls.config({ messageFormat: nls.MessageFormat.bundle, bundleFormat: nls.BundleFormat.standalone })();
+const localize: nls.LocalizeFunc = nls.loadMessageBundle();
+const noKitSelected = localize('no.kit.selected', '[No Kit Selected]');
+const noConfigPresetSelected = localize('no.configure.preset.selected', '[No Configure Preset Selected]');
+const noBuildPresetSelected = localize('no.build.preset.selected', '[No Build Preset Selected]');
+const noTestPresetSelected = localize('no.test.preset.selected', '[No Test Preset Selected]');
+
+let treeDataProvider: TreeDataProvider;
+
+export class ProjectStatus {
```

Comment:
I recommend renaming this file `projectStatus.ts`.  We have two panels in the sidebar so we probably shouldn't use the generic name for the file.

Reply:
(the other one is `projectOutlineProvider.ts` because the class was named that way. You may want to rename this one or that one to get them the names consistent with each other at some point).

Reply:
makes sense! I've updated the name